pallets/exchange/src/lib.rs

```diff
@@ -190,8 +201,16 @@ decl_module! {
 					trade_limit: min_bought
 			};
 
+			// REVIEW: This conflicts with your stated invariant `asset_a < asset_b`. Updated the doc comment above.
+			// Also the clone seems unnecessary (can be optimized by
+			// reordering the code).
+			// REVIEW: This is an unbounded append. There is nothing
+			// keeping users from inserting a lot of items here. You will want to
+			// make sure that the maximum amount of items appended in a block is ok.
+			// It's probably generally ok as the item is removed in `on_finalize`.
```

Comment:
This should be ok. This extrinsic has certain weight so it can be called only so many times to fill one block. So that would be mean, there can be only so many intentions appended. Or am i missing something ?

Reply:
Yeah, you just want to make sure that the maximum amount of transactions possible in a block is fine, size wise. Related to the benchmarking mentioned in the other comment.